src/mnode/mnode-pastel.cpp

```diff
@@ -1382,6 +1442,50 @@ std::vector<CArtTradeTicket> CArtTradeTicket::FindAllTicketByArtTnxID(const std:
     return masterNodeCtrl.masternodeTickets.FindTicketsByMVKey<CArtTradeTicket>(artTnxID);
 }
 
+std::vector<CArtTradeTicket> CArtTradeTicket::FindAllTicketByRegTnxID(const std::string& nftRegTnxId)
+{
+    return masterNodeCtrl.masternodeTickets.FindTicketsByMVKey<CArtTradeTicket>(nftRegTnxId);
+}
+
+std::map<std::string, std::string> CArtTradeTicket::GetPastelIdAndTxIdWithTopHeightPerCopy(const std::vector<CArtTradeTicket> & filteredTickets)
+{
+    //The list is already sorted by height (from beginning to end)
+    
+    //This will hold all the owner / copies serial number where serial number is the key 
+    std::map<std::string, std::string> ownerPastelIDs_and_txids;
+
+    //Copy number and winning index (within the vector)
+    std::map<std::string, int> copyOwner_Idxs;
+    int winning_idx = 0;
+    
+    for (auto & element : filteredTickets) {
+      
+      std::string serial = element.GetCopySerialNr();
+      if(copyOwner_Idxs.find(serial) != copyOwner_Idxs.end())
+      {
+        //We do have it in our copyOwner_Idxs
+        if(element.GetBlock() >= copyOwner_Idxs[serial])
```

Comment:
why do you compare block height with the saved current index of element in a vector? you should save additionally a block height for currently winning ticket and compare with it.

Reply:
There is no such thing as 'wining ticket'. Here we have 'winning tickets' (/ copy).
It is not the index what is saved, but the copy number. (or serial nr.)

Reply:
ok. and you compare index of filteredTickets that you save in copyOwner_Idxs with block height.

Reply:
Yes I compare but I save it into a map with copy_serial as key and index as value (as winning index in the filtered tickets per the given serial).